tests/datasets/test__init__.py

```diff
@@ -0,0 +1,424 @@
+
+    path_project_parent_directory.assert_called_once_with("project-path")
+    object_client.create_directory.assert_called_once_with(
+        project_id, "/", parents=True
+    )
+    os_path_isdir_mock.assert_called_once_with("local-path")
+    upload_mock.assert_called_once_with(
+        object_client, project_id, "project-path", "local-path"
+    )
+
+
+def test_datasets_put_directory(mocker, mock_client):
+    object_client, project_id = mock_client
+
+    path_project_parent_directory = mocker.patch(
+        "faculty.datasets.path.project_parent_directory", return_value="/"
+    )
+    object_client.create_directory.return_value = mocker.Mock()
+
+    os_path_isdir_mock = mocker.patch(
+        "os.path.isdir", side_effect=[True, False]
+    )
+
+    entry_mock = "test-file"
+    os_listdir_mock = mocker.patch("os.listdir", return_value=[entry_mock])
+
+    _put_file_mock = mocker.patch(
+        "faculty.datasets._put_file", return_value=mocker.Mock()
+    )
+
+    datasets.put("local-path", "project-path", project_id)
+
+    path_project_parent_directory.assert_called_once_with("project-path")
+    object_client.create_directory.assert_has_calls(
+        [
+            mocker.call(project_id, "/", parents=True),
+            mocker.call(project_id, "project-path"),
+        ]
+    )
+    os_path_isdir_mock.assert_has_calls(
+        [mocker.call("local-path"), mocker.call("local-path/test-file")]
+    )
+    os_listdir_mock.assert_called_once_with("local-path")
+    _put_file_mock.assert_called_once_with(
+        "local-path/test-file",
+        "project-path/test-file",
+        project_id,
+        object_client,
+    )
+
+
+def test_datasets_cp_path(mocker, mock_client):
+    object_client, project_id = mock_client
+    object_client.copy.return_value = mocker.Mock()
+
+    path_project_parent_directory = mocker.patch(
+        "faculty.datasets.path.project_parent_directory", return_value="/"
+    )
+    object_client.create_directory.return_value = mocker.Mock()
+
+    datasets.cp(
+        "source-path",
+        "destination-path",
+        project_id=project_id,
+        recursive=True,
+    )
+
+    path_project_parent_directory.assert_called_once_with("destination-path")
+    object_client.create_directory.assert_called_once_with(
+        project_id, "/", parents=True
+    )
+    object_client.copy.assert_called_once_with(
+        project_id, "source-path", "destination-path", recursive=True
+    )
+
+
+def test_datasets_rm_path(mocker, mock_client):
+    object_client, project_id = mock_client
+    object_client.delete.return_value = mocker.Mock()
+
+    datasets.rm("project-path", project_id=project_id, recursive=True)
+
+    object_client.delete.assert_called_once_with(
+        project_id, "project-path", recursive=True
+    )
+
+
+def test_datasets_rmdir_path(mocker):
+    project_id = mocker.Mock()
+
+    rm_mock = mocker.patch("faculty.datasets.rm", return_value=mocker.Mock())
```

Comment:
`return_value=mocker.Mock()` is the default.
```suggestion
    rm_mock = mocker.patch("faculty.datasets.rm")
```